docs/configuration.rst

```diff
@@ -19,11 +19,21 @@ For example:
    :language: python
 
 
+In addition to a list of sources, you can provide a ``doc``. You can use this to
+guide users trying to use your software and hitting configuration errors to
+documentation for your configuration.
+
+For example:
+
+.. literalinclude:: code/configuration_doc.py
```

Comment:
When I read documentation, I'm always happy to see example of what things will look like. Here I'd love to see an example of an error (like you have below) with this specific doc added to it. It makes it much easier for users to really understand what this thing is.